Abh extend autoexclude paths

ABaldwinHunter opened the pull request:
@codeclimate/review @brynary This PR adds additional directories to be autoexcluded in `codeclimate init`.

This PR adds `db`, `config`, `features`, `spec`, `test`.

Adds two additional directories to test case in spec.

noahd1 commented on lib/cc/cli/init.rb:
```diff
@@ -26,13 +26,19 @@ def create_codeclimate_yaml
       end
 
+      def auto_exclude_dir_names
+        ["config", "db", "features", "spec", "test", "vendor"]
```
I would make this a constant

noahd1 commented on lib/cc/cli/init.rb:
```diff
@@ -26,13 +26,19 @@ def create_codeclimate_yaml
+      end
+
+      def autoexclude_paths
+        @autoexclude_paths ||= auto_exclude_dir_names.map { |dir| "#{dir}/**/*" if filesystem.exist?(dir) }.compact
```
I think this would read better on multiple lines

noahd1 commented on lib/cc/cli/init.rb:
```diff
@@ -26,13 +26,19 @@ def create_codeclimate_yaml
-        if filesystem.exist?("vendor")
-          config["exclude_paths"] = ["vendor/**/*"]
-        end
+        config["exclude_paths"] = autoexclude_paths
```
I think it might read a little cleaner if you call the method `exclude_paths`  and have it take in an argument `AUTO_EXCLUDE_PATHS`

ABaldwinHunter commented:
@codeclimate/review @noahd1 thanks for feedback. I added your suggestions:

1) made a constant `AUTO_EXCLUDE_PATHS`
2) refactored `exclude_paths` method to span multiple lines
3) and take `AUTO_EXCLUDE_PATHS` as an argument

GordonDiggs commented on lib/cc/cli/init.rb:
```diff
@@ -3,6 +3,8 @@ module CLI
     class Init < Command
       include CC::Analyzer
 
+      AUTO_EXCLUDE_PATHS = ["config", "db", "features", "spec", "test", "vendor"].freeze
```
:lipstick: use `%w[]` for word arrays

GordonDiggs commented:
:+1: LGTM
